vaadin-select-flow/src/main/java/com/vaadin/flow/component/select/data/SelectDataViewImpl.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2000-2020 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.vaadin.flow.component.select.data;
+
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import com.vaadin.flow.component.ComponentEventListener;
+import com.vaadin.flow.component.ComponentUtil;
+import com.vaadin.flow.component.select.Select;
+import com.vaadin.flow.data.provider.DataProvider;
+import com.vaadin.flow.data.provider.Query;
+import com.vaadin.flow.data.provider.SizeChangeEvent;
+import com.vaadin.flow.function.SerializableSupplier;
+import com.vaadin.flow.shared.Registration;
+
+/**
+ * Implementation of the base DataView applicable for use with any
+ * DataProvider.
+ * Note that this has a minimal API and using {@link SelectListDataView}
+ * is advised.
+ *
+ * @param <T>
+ *         item type
+ * @since
+ */
+public class SelectDataViewImpl<T> implements SelectDataView<T> {
```

Comment:
Why can't this `extend AbstractDataView` ?

It is kind of the same question for `GridDataViewImpl` but it at least requires `DataCommunicator` so I thought it was fine there.

Reply:
Thinking only of validating against given dataprovider type as required and getting stuck on the `protected final void` method for validation.

Fixed also grid would be fine with it also.